cpp/src/arrow/compute/kernels/scalar_arithmetic_test.cc

```diff
@@ -492,6 +486,81 @@ TYPED_TEST(TestBinaryArithmeticFloating, Add) {
   this->AssertBinop(Add, "[null, 2.0]", this->MakeNullScalar(), "[null, null]");
 }
 
+TYPED_TEST(TestBinaryArithmeticFloating, Div) {
+  for (auto check_overflow : {false, true}) {
+    this->SetOverflowCheck(check_overflow);
+    this->AssertBinop(Divide, "[]", "[]", "[]");
+
+    this->AssertBinop(Divide, "[3.4, 2.6, 6.3]", "[1, 2, 2]", "[3.4, 1.3, 3.15]");
+
+    this->AssertBinop(Divide, "[1.1, 2.4, 3.5, 4.3, 5.1, 6.8, 7.3]",
+                      "[1.0, 2.0, 0.7, 0.5, 1.7, 2.0, 5.0]",
+                      "[1.1, 1.2, 5.0, 8.6, 3.0, 3.4, 1.46]");
```

Comment:
It's not useful to test so many values. We know the CPU handles division properly. Instead, focus on test cases that may trigger specific behaviour:
* empty array
* nulls
* zeros
* for floating-point: infinities, perhaps not-a-number

Reply:
Thank you for the good suggestion. 
I have removed some test cases, and made sure the remaining cases cover the above items, with one exception: the inifinity comparisons should depend on another PR: https://github.com/apache/arrow/pull/7826

Reply:
#7826 was merged now. Also, the tested values are still plethoric IMHO, which makes the tests less readable.

Reply:
Thanks a lot for your effort. I have added test case for infinity. 
In addition, I have removed/simplified more test cases, please check if it looks good to you.